src/interactive-auth.ts

```diff
@@ -163,7 +163,7 @@ export class NoAuthFlowFoundError extends Error {
  *
  * The generic parameter `T` is the type of the response of the endpoint, once it is eventually successful.
  */
-export type UIAuthCallback<T> = (makeRequest: (authData: IAuthDict) => Promise<UIAResponse<T>>) => Promise<T>;
+export type UIAuthCallback<T> = (makeRequest: (authData: IAuthDict | null) => Promise<UIAResponse<T>>) => Promise<T>;
```

Comment:
please can we document what it means when `authData` is null? What is `makeRequest`'s behaviour in that case? (And why do we have `UIAuthCallback` implementations that set it to `null` rather than a real `IAuthDict`?)